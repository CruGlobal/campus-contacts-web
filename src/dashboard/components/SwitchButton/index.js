import styled from '@emotion/styled';
import React, { useState } from 'react';
import { withTheme } from 'emotion-theming';
import PropTypes from 'prop-types';

const Container = styled.div`
    width: 145px;
    height: 32px;
    border-radius: 16px;
    overflow: hidden;
    display: flex;
`;

const Button = styled.div`
    width: 72px;
    background-color: #eceef2;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${({ theme }) => theme.colors.secondary};
    font-size: 14px;
    line-height: 20px;

    &:not(:last-child) {
        margin-right: 1px;
    }

    &:hover {
        background-color: #e2e2e2;
        cursor: pointer;
    }

    &.active {
        background-color: ${({ theme }) => theme.colors.highlight};
        color: ${({ theme }) => theme.colors.white};
    }
`;

const SwitchButton = ({
    leftLabel,
    rightLabel,
    onLeftClick,
    onRightClick,
    isMonth,
}) => {
    const [isLeftActive, setLeftActive] = useState(isMonth);

    const leftClicked = () => {
        setLeftActive(true);
        onLeftClick();
    };

    const rightClicked = () => {
        setLeftActive(false);
        onRightClick();
    };

    return (
        <Container>
            <Button
                className={isLeftActive ? 'active' : ''}
                onClick={leftClicked}
            >
                {leftLabel}
            </Button>
            <Button
                className={isLeftActive ? '' : 'active'}
                onClick={rightClicked}
            >
                {rightLabel}
            </Button>
        </Container>
    );
};

export default withTheme(SwitchButton);

SwitchButton.propTypes = {
    leftLabel: PropTypes.string,
    rightLabel: PropTypes.string,
    onLeftClick: PropTypes.func,
    onRightClick: PropTypes.func,
};
