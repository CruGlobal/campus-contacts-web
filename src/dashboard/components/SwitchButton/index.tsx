import styled from '@emotion/styled';
import React, { useState } from 'react';
import { withTheme } from 'emotion-theming';

const Container = styled.div`
    width: 145px;
    height: 32px;
    border-radius: 16px;
    overflow: hidden;
    display: flex;
`;

const Button = styled.div`
    width: 72px;
    background-color: #eceef2;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${({ theme }: { theme: any }) => theme.colors.secondary};
    font-size: 14px;
    line-height: 20px;

    &:not(:last-child) {
        margin-right: 1px;
    }

    &:hover {
        background-color: #e2e2e2;
        cursor: pointer;
    }

    &.active {
        background-color: ${({ theme }: { theme: any }) =>
            theme.colors.highlight};
        color: ${({ theme }: { theme: any }) => theme.colors.white};
    }
`;

interface Props {
    leftLabel: string;
    rightLabel: string;
    onLeftClick: () => void;
    onRightClick: () => void;
    isMonth: boolean;
}

const SwitchButton = ({
    leftLabel,
    rightLabel,
    onLeftClick,
    onRightClick,
    isMonth,
}: Props) => {
    const [isLeftActive, setLeftActive] = useState(isMonth);

    const leftClicked = () => {
        setLeftActive(true);
        onLeftClick();
    };

    const rightClicked = () => {
        setLeftActive(false);
        onRightClick();
    };

    return (
        <Container>
            <Button
                className={isLeftActive ? 'active' : ''}
                onClick={leftClicked}
            >
                {leftLabel}
            </Button>
            <Button
                className={isLeftActive ? '' : 'active'}
                onClick={rightClicked}
            >
                {rightLabel}
            </Button>
        </Container>
    );
};

export default withTheme(SwitchButton);
