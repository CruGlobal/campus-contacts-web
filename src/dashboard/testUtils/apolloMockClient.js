import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
