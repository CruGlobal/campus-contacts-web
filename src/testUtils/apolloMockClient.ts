import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
