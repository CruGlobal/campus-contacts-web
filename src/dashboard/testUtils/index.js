import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
