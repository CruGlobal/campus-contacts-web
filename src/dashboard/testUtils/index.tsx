import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
