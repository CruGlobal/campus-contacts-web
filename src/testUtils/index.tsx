import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
