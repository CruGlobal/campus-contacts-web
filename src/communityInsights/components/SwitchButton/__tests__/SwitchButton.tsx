import React from 'react';

import { renderWithContext } from '../../../../testUtils';

import SwitchButton from '..';

describe('<SwitchButton />', () => {
    it('should render properly', async () => {
        renderWithContext(
            <SwitchButton
                leftLabel={'left'}
                rightLabel={'right'}
                onLeftClick={() => {}}
                onRightClick={() => {}}
                isMonth={true}
            />,
        ).snapshot();
    });
});
