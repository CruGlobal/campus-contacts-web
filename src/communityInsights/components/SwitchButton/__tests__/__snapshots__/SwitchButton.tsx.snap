// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchButton /> should render properly 1`] = `
.emotion-2 {
  width: 145px;
  height: 32px;
  border-radius: 16px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  width: 72px;
  background-color: #eceef2;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #B4B6BA;
  font-size: 14px;
  line-height: 20px;
}

.emotion-0:not(:last-child) {
  margin-right: 1px;
}

.emotion-0:hover {
  background-color: #e2e2e2;
  cursor: pointer;
}

.emotion-0.active {
  background-color: #3CC8E6;
  color: #ffffff;
}

<div>
  <div
    class="emotion-2"
  >
    <div
      class="active emotion-0"
    >
      left
    </div>
    <div
      class=" emotion-0"
    >
      right
    </div>
  </div>
</div>
`;
